packages/product-badge/style.module.css

```diff
@@ -0,0 +1,98 @@
+.root {
+  display: flex;
+  align-items: center;
+  height: 24px;
+  padding: 1.5px 12px 2.5px 12px;
+  border-radius: 24px;
+  width: max-content;
+
+  &.light {
+    &.primary {
+      background-color: var(--brand);
+    }
+
+    &.secondary {
+      background-color: var(--wpl-neutral-50);
+    }
+
+    &.hasDot {
+      background-color: var(--wpl-neutral-50);
+    }
+  }
+
+  &.dark {
+    &.primary {
+      background-color: var(--wpl-neutral-0);
+    }
+    &.secondary {
+      background-color: var(--wpl-neutral-700);
+    }
+    &.hasDot {
+      background-color: var(--wpl-neutral-700);
+    }
```

Comment:
💭 Thoughts on DRYing this up a bit with CSS variables?

- We're in a flex container so use gap vs absolute positioning to space dot
- Reduce dupe ::before CSS by moving to root
- Use currrentColor to inherit the text color

```css
.root {
  --color: var(--brand-text-on-primary);
  --background-color: var(--brand);

  color: var(--color);
  background-color: var(--background-color);
  display: flex;
  align-items: center;
  gap: 5px;
  height: 24px;
  padding: 1.5px 12px 2.5px 12px;
  border-radius: 24px;
  width: max-content;

  &::before {
    --size: 7px;

    content: '';
    flex-shrink: 0;
    width: var(--size);
    height: var(--size);
    background-color: var(--brand);
    border-radius: 9999px;
  }
}

.light {
  &.secondary,
  &.hasDot {
    --background-color: var(--wpl-neutral-50);
    --color: var(--wpl-neutral-500);
  }
}

.dark {
  &.secondary,
  .hasDot {
    --background-color: var(--wpl-neutral-700);
    --color: var(--wpl-neutral-200);
  }
}

.text {
  composes: g-type-label from global;
  position: relative;
  text-transform: capitalize;
  margin: 0;
  color: currentColor;
}
```

Reply:
Updated!